spec/connect/client_spec.rb

```diff
@@ -287,51 +287,75 @@
 
       allow(subject).to receive(:show_product).and_return product
       allow(subject).to receive(:register_product).and_return true
+      allow(subject).to receive(:announce_system)
       SUSE::Connect::GlobalLogger.instance.log = string_logger
     end
 
     after do
       SUSE::Connect::GlobalLogger.instance.log = default_logger
     end
 
-    it 'should call announce if system not registered' do
-      expect(subject).to receive(:announce_system)
-      subject.register!
-    end
+    context 'when the system is not registered' do
+      it 'announces the system' do
+        expect(subject).to receive(:announce_system)
+        subject.register!
+      end
 
-    it 'should not call announce but update on api if system registered' do
-      allow(System).to receive_messages(credentials?: true)
-      expect(subject).not_to receive(:announce_system)
-      expect(subject).to receive(:update_system)
-      subject.register!
-    end
+      it 'writes credentials file' do
+        allow(System).to receive_messages(credentials?: false)
+        allow(subject).to receive_messages(announce_system: %w[lg pw])
+        expect(Credentials).to receive(:new).with('lg', 'pw', Credentials::GLOBAL_CREDENTIALS_FILE).and_call_original
```

Comment:
It would be nice not to `call_original`, but return a double of `Credentials` instead.

Reply:
Fixed!